File: 00-node/01-examples/01-get-with-params/core/book.js
'use strict'

let book = {
    getBook: (id) => {
        let books = [
            {id: '1', title: 'Code Complete 2nd Edition', author: 'Steve McConnell', year: '2004'}, 
            {id: '2', title: 'Clean Code', author: 'Robert C. Martin', year: '2008'}, 
            {id: '3', title: 'The Last Lecture', author: 'Randy Pausch', year: '2008'} 
        ];

        let filteredBook = books.find(b => b.id === id); 

        return filteredBook;
    }
};

module.exports = book;

File: 02-netcore/01-examples/01-get-with-params/Core/Book.cs
using System.Collections.Generic;

using _01_get_with_params.Core.Models;

namespace _01_get_with_params.Core
{
    public class Book
    {
        public BookInformation GetBook(string bookId)
        {
            var books = new List<BookInformation>();
            books.Add(new BookInformation("1", "Code Complete 2nd Edition", "Steve McConnell","2004"));
            books.Add(new BookInformation("2", "Clean Code", "Robert C. Martin","2008"));
            books.Add(new BookInformation("3", "The Last Lecture", "Randy Pausch","2008"));
            
            var filteredBook = books.Find(b => b.Id == bookId);

            return filteredBook;
        }

    }
}